database/src/main/scala/com/rasterfoundry/granary/TokenDao.scala

```diff
@@ -0,0 +1,39 @@
+package com.rasterfoundry.granary.database
+
+import cats.implicits._
+import doobie._
+import doobie.implicits._
+import doobie.postgres.implicits._
+
+import java.util.UUID
+
+object TokenDao {
+  sealed abstract class TokenDaoError extends Throwable
+  case object InvalidToken            extends TokenDaoError
+
+  val selectF =
+    fr"""
+  SELECT id
+  FROM tokens
+  """
+
+  def validateToken(
+      id: String
+  ): ConnectionIO[Either[TokenDaoError, Boolean]] =
+    try {
+      val uuid = UUID.fromString(id)
```

Comment:
The `try` block should cover just the error-y part -- we can also throw `UUID.fromString` into an IO and use `LiftIO` to get it into a `ConnectionIO`, then do everything in the same monadic context, e.g. https://github.com/raster-foundry/raster-foundry/blob/7045056b64235f86da1553f47a1b514611951b25/app-backend/db/src/main/scala/ProjectLayerDao.scala#L141-L159 (though you can `LiftIO[ConnectionIO].liftIO`, instead of asking for the `implicit L` as we do there -- I think I'm also responsible for that pattern :man_facepalming: )